core/src/main/java/org/geysermc/geyser/configuration/GeyserJacksonConfiguration.java

```diff
@@ -139,6 +139,9 @@
     @JsonProperty("xbox-achievements-enabled")
     private boolean xboxAchievementsEnabled = false;
 
+    @JsonProperty("custom-model-data-enabled")
+    private boolean customModelDataEnabled = true;
```

Comment:
Why do we need a config option for this? Either you want to load the mappings, or you don't have mappings. I'd gate this under `addNonBedrockItems`.